bindings/rust/s2n-tls/src/connection.rs

```diff
@@ -61,9 +61,56 @@ impl fmt::Debug for Connection {
 
 /// # Safety
 ///
-/// s2n_connection objects can be sent across threads
+/// NonNull / the raw s2n_connection pointer isn't Send because its data may be
```

Comment:
This is a note in general for your doc comments. It's not important to know why the raw pointer isn't Send/Sync. I would prefer it if you started out with why the Connection/Config/CertChain _is_ Send/Sync. If you need to elaborate on why the raw s2n_connection pointer isn't Send/Sync, maybe do that after? But I think most people are just looking for why this Rust struct was tagged with Send/Sync.

So with that in mind, I would probably restructure your comment like:
The Connection is Send because the existing interface ensures that only one owned Connection can exist for each s2n_connection C object. 
The raw s2n_connection pointer it wraps is not Send, and therefore library developers MUST ensure that new methods do not expose the raw s2n_connection pointer...